File: Sample/ApplicationSample/sample_cs/MapObject2D/MapObject2D_Basic.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/// <summary>
/// マップを表示するサンプル。
/// </summary>
class MapObject2D_Basic : ISample
{
	public string Description
	{
		get { return "マップチップを並べて描画するサンプルです。"; }
	}

	public string Title
	{
		get { return "マップチップの描画"; }
	}

	public string ClassName
	{
		get { return "MapObject2D_Basic"; }
	}

	public void Run()
    {
        // Altseedを初期化する。
        asd.Engine.Initialize("MapObject2D_Basic", 640, 480, new asd.EngineOption());

        {
            //マップオブジェクトを生成する。
            var mapObject = new asd.MapObject2D();

            var texture = asd.Engine.Graphics.CreateTexture2D("Data/Texture/Chip1.png");

            //マップオブジェクトに16*12=184個のチップを登録する。
            for (int i = 0; i < 16; ++i)
            {
                for (int j = 0; j < 12; ++j)
                {
                    //チップを生成する。
                    var chip = new asd.Chip2D();

                    //チップにテクスチャを設定する。
                    chip.Texture = texture;

                    //チップの描画先を指定する。
                    chip.Position = new asd.Vector2DF(i * 40, j * 40);

                    //マップオブジェクトにチップを追加する。
                    mapObject.AddChip(chip);
                }
            }

            //レイヤーにマップオブジェクトを追加する。
            asd.Engine.AddObject2D(mapObject);
        }

        // Altseedのウインドウが閉じられていないか確認する。
        while (asd.Engine.DoEvents())
        {
            // Altseedを更新する。
            asd.Engine.Update();
            Recorder.TakeScreenShot("MapObject2D_Basic", 30);
        }

        // Altseedの終了処理をする。
        asd.Engine.Terminate();
    }
}

File: Sample/ApplicationSample/sample_cpp/MapObject2D/MapObject2D_Basic.cpp

#include <Base.h>

/**
@brief	マップを表示するサンプル。
*/
void MapObject2D_Basic()
{
	// Altseedを初期化する
	asd::Engine::Initialize(u"MapObject2D_Basic", 640, 480, asd::EngineOption());

	{
		//マップオブジェクトを生成する。
		auto mapObject = std::make_shared<asd::MapObject2D>();

		auto texture = asd::Engine::GetGraphics()->CreateTexture2D(u"Data/Texture/Chip1.png");

		//マップオブジェクトに16*12=184個のチップを登録する。
		for (int i = 0; i < 16; ++i)
		{
			for (int j = 0; j < 12; ++j)
			{
				//チップを生成する。
				auto chip = std::make_shared<asd::Chip2D>();

				//チップにテクスチャを設定する。
				chip->SetTexture(texture);

				//チップの描画先を指定する。
				chip->SetPosition(asd::Vector2DF(i * 40, j * 40));

				//マップオブジェクトにチップを追加する。
				mapObject->AddChip(chip);
			}
		}

		// マップオブジェクトのインスタンスをエンジンへ追加する。
		asd::Engine::AddObject2D(mapObject);
	}

	// Altseedのウインドウが閉じられていないか確認する。
	while (asd::Engine::DoEvents())
	{
		// Altseedを更新する。
		asd::Engine::Update();
	}

	// Altseedを終了する。
	asd::Engine::Terminate();
}